src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -348,9 +351,42 @@ void PCRPrimerDesignForDNAAssemblyOPWidget::sl_onFindTaskFinished() {
     CHECK(pcrTask->isFinished(), );
     productsTable->setCurrentProducts(pcrTask->getResults(), annDnaView);
     createResultAnnotations();
+    backboneSequence = pcrTask->getBackboneSequence();
+    lastRunSettings = pcrTask->getSettings();
     pcrTask = nullptr;
 }
 
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_extractProduct() {
+    ADVSequenceObjectContext *sequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(NULL != sequenceContext, L10N::nullPointerError("Sequence Context"), );
+    U2SequenceObject *sequenceObject = sequenceContext->getSequenceObject();
+    SAFE_POINT(NULL != sequenceObject, L10N::nullPointerError("Sequence Object"), );
+    ExtractPrimerTaskSettings settings;
+    settings.sequenceRef = sequenceContext->getSequenceRef();
+    GUrl sequenceURL = sequenceContext->getSequenceObject()->getDocument()->getURL();
+    if (sequenceURL.getType() == GUrl_File) {
+        settings.originalSequenceFileName = QFileInfo(sequenceURL.getURLString()).baseName();
+    } else {
+        settings.originalSequenceFileName = sequenceContext->getSequenceObject()->getSequenceName();
+    }
+    if (lastRunSettings.insertTo == PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone5) {
+        settings.direction = ExtractPrimerTaskSettings::FiveToFive;
+        settings.backboneLength = lastRunSettings.bachbone5Length;
+    } else {
+        CHECK(lastRunSettings.insertTo == PCRPrimerDesignForDNAAssemblyTaskSettings::BackboneBearings::Backbone3, )
+        settings.direction = ExtractPrimerTaskSettings::FiveToThree;
+        settings.backboneLength = lastRunSettings.bachbone3Length;
+    }
+    settings.backboneSequence = backboneSequence;
+    Annotation *selectedAnnotation = productsTable->getSelectedAnnotation();
+    if (selectedAnnotation != nullptr) {
+        CHECK(selectedAnnotation->getRegions().size() == 1, );
```

Comment:
CHECK of SAFE_POINT?

Reply:
As i can see, it's CHECK. Better to tell me what you want here.

Reply:
As i understand CHECK is warning for developer, but SAFE_POINT is for user. And this verification definitely is not for user.

Reply:
`CHECK` is a legal situation, which could really happen.
`SAFE_POINT` is a kind of situations, which shouldn't happen. 

So, if the condition could fail from time to time and it wont break the application - it's `CHECH`. If the condition should ALWAYS be true  and couldn't be `false` - it's `SAFE_POINT`